### PR TITLE
[bitnami/external-dns] Fix missing pod labels in Deployment/DaemonSet

### LINKED ISSUE #18827: [bitnami/external-dns] Upgrading to 6.24.1 removes the defined podLabels from the deployment
### Name and Version

bitnami/external-dns 6.24.1

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Define podLables
2. Deploy external-dns helm chart in version 6.23.4
3. Upgrade to 6.24.1

### Are you using any custom parameters or values?

image

### What is the expected behavior?

Defined podLabels should be set

### What do you see instead?

Values set in podLabels getting removed

image

### Additional information

Guessing this commit caused the issue:
https://github.com/bitnami/charts/commit/a0f8f0ea81c60a0a915424184adfc031acf475f8

podLabels are missing in the deployment:

https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/dep-ds.yaml#L15

### PR DESCRIPTION
**Pull request by Mahagon**
(#18827)

### Description of the change

.Values.podLabels weren't applied to the deployment/daemonset of external-dns

### Applicable issues

- fixes #18827

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/templates/dep-ds.yaml [hunk 1]
```diff
@@ -12,7 +12,8 @@ kind: Deployment
 metadata:
   name: {{ template "external-dns.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
-  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- $podLabels := merge .Values.podLabels .Values.commonLabels }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" $podLabels "context" $ ) | nindent 4 }}
```
Comment (juan131) [inline bitnami/external-dns/templates/dep-ds.yaml:16]: Please note that these aren't pod labels but sts/deployment labels. Therefore, we shouldn't apply `.Values.podLabels` here

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for you contribution! Please check my comments

File: bitnami/external-dns/templates/dep-ds.yaml (hunk 1 above)
Comment (Mahagon) [reply bitnami/external-dns/templates/dep-ds.yaml:16]: Ah you are completely right
i was fooled by the helmfile diff which told me the labels were removed. but they were just removed from the deployment

**Comment by Mahagon** [comment]
It's working fine, podlabels were just removed from the deployment/daemonset

**Pull request closed without merging**